File: java/pe002.java
//file: pe002.java

public class pe002 {
  public static void main( String[] args ) {
    final int max = 4000000;
    int current = 1;
    int previous = 1;
    int sum = 0;
    while (current < max) {
      int next = current + previous;
      if ((current % 2) == 0) {
        sum += current;
      }
      previous = current;
      current = next;
    }
    System.out.println(sum);
  }
}

File: elixir/pe002.exs
defmodule ProjectEuler do
  def pe002(n) do
    sum_even_fibonacci(1, 1, n, 0)
  end

  defp sum_even_fibonacci(previous, current, n, sum) when current < n and rem(current, 2) == 0 do
    sum_even_fibonacci(current, previous + current, n, sum + current)
  end

  defp sum_even_fibonacci(previous, current, n, sum) when current < n do
    sum_even_fibonacci(current, previous + current, n, sum)
  end

  defp sum_even_fibonacci(_, _, _, sum) do
    sum
  end
end

IO.puts(ProjectEuler.pe002(4_000_000))
